docs/cpm.rst

```diff
@@ -0,0 +1,253 @@
+CPM Reference
+#############
+
+The `rapids_cpm` functions allow projects to find or build dependencies with built-in
+tracking of these dependencies for correct export support.
+
+rapids-cmake package defaults
```

Comment:
This sequence is a bit odd. I would prefer
1. rapids-cmake provides packages
2. rapids-cmake encodes information about package versions in a json file; mention that it includes some set of defaults (but don't list here) then go into the format itself
3. overrides
4. pinning and reproducible builds
5. list of defaults

Reply:
The primary reader I had in mind for this document was a consumer of rapids-cmake and not a developer. So I wanted to first cover what cpm packages we offer, and the features ( overrides, pins ) that interact with the versions.json.

Therefore in those cases the exact details on the versions file are last, since that is the most technical detail that isn't needed unless you are writing your own override

Reply:
When you say developers, who do you mean? I assume that all consumers of rapids-cmake are developers of other libraries, no?

But assuming I'm misunderstanding the audience you intend, I agree that the sequence seems to make sense for someone who doesn't need to know about overrides.

Reply:
>  I agree that the sequence seems to make sense for someone who doesn't need to know about overrides.

:+1: 

> When you say developers, who do you mean? I assume that all consumers of rapids-cmake are developers of other libraries, no?

Downstream build-system developers was whom I was thinking about as 'developers'.